src/ovs/aca_ovs_programmer.cpp

```diff
@@ -31,6 +32,48 @@ extern bool g_demo_mode;
 
 namespace aca_ovs_programmer
 {
+static int aca_set_port_vlan_workitem(const string port_name, uint vlan_id)
+{
+  ACA_LOG_DEBUG("aca_set_port_vlan_workitem ---> Entering\n");
+
+  ulong not_care_culminative_time = 0;
+  int overall_rc;
+
+  if (port_name.empty()) {
+    throw std::invalid_argument("port_name is empty");
+  }
+
+  if (vlan_id == 0) {
+    throw std::invalid_argument("vlan_id is zero");
+  }
+
+  static ushort MAX_PORT_WAIT_SECONDS = 300; // 5 mins
+  uint waited_seconds = 0;
+  string cmd_string = "set port " + port_name + " tag=" + to_string(vlan_id);
+
+  do {
+    std::this_thread::sleep_for(chrono::milliseconds(1000));
+
+    overall_rc = EXIT_SUCCESS;
+    ACA_OVS_Programmer::get_instance().execute_ovsdb_command(
+            cmd_string, not_care_culminative_time, overall_rc);
+
+    if (overall_rc == EXIT_SUCCESS)
+      break;
+  } while (++waited_seconds < MAX_PORT_WAIT_SECONDS);
```

Comment:
I would recommend to use retry_time instead of waited_second.

Reply:
fixed